cirq-core/cirq/sim/simulation_utils.py

```diff
@@ -56,4 +56,4 @@ def state_probabilities_by_indices(
         probs = np.sum(probs, axis=-1)
 
     # To deal with rounding issues, ensure that the probabilities sum to 1.
-    return probs / np.sum(probs)
+    return np.clip(probs / np.sum(probs), 0, 1)
```

Comment:
To satisfy the summation to 1 it is more stable to first clip and then normalize -
```suggestion
    probs.clip(0, None, out=probs)
    probs /= probs.sum()
    return probs
```